corehq/apps/app_manager/templates/app_manager/form_view_base.html

```diff
@@ -103,9 +103,18 @@
             var isModuleFilterEnabled = {{ is_module_filter_enabled|BOOL }};
             var formFilter = {{ form.form_filter|JSON }};
             var allOtherFormsRequireACase = {{ form.all_other_forms_require_a_case|BOOL }};
+            var formFilterReferencesCase = {{ form_filter_references_case|BOOL }};
             var formFilterAllowed = ko.computed(function () {
```

Comment:
so actually that's what's causing the issue in http://manage.dimagi.com/default.asp?222086 - the current behavior is that the form filter ui is always shown when module filtering is enabled, regardless of whether the forms in the module use cases or not, but then when you make the build. the backend filter always silently leaves the filters out of the suite file unless all the forms in the module use cases.

so i think we're still going to have the same issue unless we make the ui logic and the backend logic match up.  @amsagoff can you describe the way that we'd like form filtering to behave? it sounds like it's something along the lines of:

- always allow entering form filter conditions
- if a form filter condition is entered that references a case but the module is set to "show only forms" or not all of the forms in the module use a case, then raise an error during build. also an error will be displayed in the form filter ui once the page is refreshed (doing so without a refresh involves a little more work)
- if a form filter condition is entered that references a user case: not sure what the right behavior is here, and what the right criteria are for determining whether a form filter condition references a user case